apps/nccl/src/nccl.cu

```diff
@@ -524,6 +529,13 @@ static void ncclCommInitRankFallbackSingleNode(ncclComm* commPtr, std::shared_pt
   commPtr->scratchBuff = mscclpp::GpuBuffer<char>(SCRATCH_SIZE).memory();
   commPtr->remoteScratchRegMemories =
       setupRemoteMemories(commPtr->comm, rank, commPtr->scratchBuff.get(), SCRATCH_SIZE, mscclpp::Transport::CudaIpc);
+
+  hipMalloc((void**)&(commPtr->syncer), sizeof(mscclpp::DeviceSyncer));
+  hipMemset((void*)(commPtr->syncer), 0, sizeof(mscclpp::DeviceSyncer));
+
+  uint64_t initFlag = 1;
+  hipMalloc((void**)&(commPtr->deviceFlag), sizeof(uint64_t));
+  hipMemcpy((void*)(commPtr->deviceFlag), &initFlag, sizeof(uint64_t), hipMemcpyHostToDevice);
```

Comment:
Maybe use cudaMalloc here, we already map cudaMalloc, cudaFree... to hipMalloc, hipFree

Reply:
will do.